comodo2/src/comodo2/queries/QStateMachine.java

```diff
@@ -64,6 +64,28 @@ public String getInitialStateName(final StateMachine sm) {
 		return "";
 	}
 
+	/**
+	 * This method looks for the initial state of a State Machine.
+	 * It takes the "initial" pseudo-state (kind == "initial")
+	 * at the level of the SM (container.owner == sm)
+	 * and returns the destination state of the associated
+	 * transition.
+	 * @param sm State Machine
+	 * @return The initial state instance if it exists, null otherwise.
+	 */
+	public State getInitialState(final StateMachine sm) {
```

Comment:
In theory one could refactor getInitialStateName() above to use getInitialState()